### PR TITLE
HTML themes can set up default sidebars through ``theme.conf``

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- Give a better default sidebars setting from 3rd party theme (cf. alabaster)

### Detail
- Add `sidebar` setting to `theme.conf`
- It changes the default sidebar setting
- It allows to override it by `html_sidebars`

### Relates
- #4102

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -820,8 +820,9 @@ that use Sphinx's HTMLWriter class.
      to include.  If all or some of the default sidebars are to be included,
      they must be put into this list as well.
 
-     The default sidebars (for documents that don't match any pattern) are:
-     ``['localtoc.html', 'relations.html', 'sourcelink.html',
+     The default sidebars (for documents that don't match any pattern) are
+     defined by theme itself.  Builtin themes are using these templates by
+     default: ``['localtoc.html', 'relations.html', 'sourcelink.html',
      'searchbox.html']``.
```
Comment (jfbu) [inline doc/config.rst:826]: Perhaps this info should also be in quickstart created conf.py ?

File: sphinx/templates/quickstart/conf.py_t [hunk 2]
```diff
@@ -110,14 +110,7 @@ html_static_path = ['{{ dot }}static']
 # Custom sidebar templates, must be a dictionary that maps document names
 # to template names.
```
Comment (jfbu) [inline sphinx/templates/quickstart/conf.py_t:111]: (see my comment above)

**Review by jfbu — APPROVED** [review]
LGTM with nits: this does not seem to make sure alabaster theme uses ``'about.html'`` (and ``'navigation.html'``) per default, without them and using rather the same default as built-in themes the looks of Alabaster are modified and do not fit Alabaster RTD doc guidelines.

**Comment by tk0miya** [comment]
Updated.

**Comment by tk0miya** [comment]
Merged.

**Comment by alexander-schranz** [comment]
the changes here in `sphinx/themes/basic/theme.conf` does break exist themes not sure what the bc promises  are for minor releases here setting the sidebars to empty did fix the exist theme but cost me some time to find the error and not sure if thats a valid configuration?

```conf
sidebars =
```

**Pull request merged**